src/importexport/tabledit/internal/importtef.cpp

```diff
@@ -0,0 +1,1045 @@
+                keysig->setKey(Key(tefMeasure.key));
+                keysig->setTrack(i * VOICES);
+                s1->add(keysig);
+            }
+            lastKey = tefMeasure.key;
+
+            auto s2 = measure->getSegment(mu::engraving::SegmentType::TimeSig, tick);
+            for (size_t i = 0; i < tefInstruments.size(); ++i) {
+                mu::engraving::TimeSig* timesig = Factory::createTimeSig(s2);
+                timesig->setSig(length);
+                timesig->setTrack(i * VOICES);
+                s2->add(timesig);
+            }
+            lastTimeSig = length;
+            createTempo();
+        }
+        else {
+            if (tefMeasure.key != lastKey) {
+                auto s1 = measure->getSegment(mu::engraving::SegmentType::KeySig, tick);
+                for (size_t i = 0; i < tefInstruments.size(); ++i) {
+                    mu::engraving::KeySig* keysig = Factory::createKeySig(s1);
+                    keysig->setKey(Key(tefMeasure.key));
+                    keysig->setTrack(i * VOICES);
+                    s1->add(keysig);
+                }
+                lastKey = tefMeasure.key;
+            }
+            if (length != lastTimeSig) {
+                auto s2 = measure->getSegment(mu::engraving::SegmentType::TimeSig, tick);
+                for (size_t i = 0; i < tefInstruments.size(); ++i) {
+                    mu::engraving::TimeSig* timesig = Factory::createTimeSig(s2);
+                    timesig->setSig(length);
+                    timesig->setTrack(i * VOICES);
+                    s2->add(timesig);
+                }
+                lastTimeSig = length;
+            }
+        }
+
+        tick += length;
+    }
+}
+
+void TablEdit::createNotesFrame()
+{
+    if (!tefHeader.notes.empty()) {
+        VBox* vbox = Factory::createTitleVBox(score->dummy()->system());
+        vbox->setTick(mu::engraving::Fraction(0, 1)); // TODO find correct value (0/1 seems to work OK)
+        score->measures()->add(vbox);
+        Text* s = Factory::createText(vbox, TextStyleType::FRAME);
+        s->setPlainText(String::fromUtf8(tefHeader.notes.c_str()));
+        vbox->add(s);
+    }
+}
+
+void TablEdit::createParts()
+{
+    for (const auto& instrument : tefInstruments) {
+        Part* part = new Part(score);
+        score->appendPart(part);
+        String staffName { String::fromUtf8(instrument.name.c_str()) };
+        part->setPartName(staffName);
+        part->setPlainLongName(staffName);
+
+        StringData stringData;
+        stringData.setFrets(25); // reasonable default (?)
+        for (int i = 0; i < instrument.stringNumber; ++i) {
+            int pitch = 96 - instrument.tuning.at(instrument.stringNumber - i - 1);
+            LOGD("pitch %d", pitch);
+            instrString str { pitch };
+            stringData.stringList().push_back(str);
+        }
+        part->instrument()->setStringData(stringData);
```

Comment:
For playback to work correctly, we'll also need to set an instrument ID. See perhaps `searchTemplateForInstrNameList`, but maybe there are better ways in this case.